### ISSUE TITLE
Is 'droplet destroy' subcommand has bug ?

### ISSUE THREAD

**Issue by sawanoboly** [rmoriz/knife-digital_ocean#73]
https://github.com/rmoriz/knife-digital_ocean/blob/master/lib/chef/knife/digital_ocean_droplet_destroy.rb#L39-L47

Apparently, we must give `--server` to subcomand for destroy droplet. Can't use only -`-all`.
And we must wait long time before destroy single droplet with `--server` option.

It is expected this way intrinsically below?

- We can destroy a single droplet with `droplet destroy --server SERVER_ID`, and it effect soon.
- We have to use `droplet destroy --all` to destroy all servers, and wait long time until starting destroy.

**Comment by rmoriz** [comment]
#76

**Comment by rmoriz** [comment]
I currently don't have a Digital Ocean test account to verify the change and run all the tests. Any volunteers? :-)

gem 'knife-digital_ocean', github: 'rmoriz/knife-digital_ocean', branch: 'destroy-fix'

**Comment by rmoriz** [comment]
released 2.4.1

**Comment by rudijs** [comment]
Hi,

Still an issue for me using version 2.4.1

```
knife digital_ocean droplet destroy --server 5966659 -VV 
INFO: Using configuration from /media/rudi/crypt2/projects/ride-share-market/devops/app/kitchen/.chef/knife.rb
/home/rudi/.rvm/gems/ruby-2.2.0/gems/knife-digital_ocean-2.4.1/lib/chef/knife/digital_ocean_droplet_destroy.rb:50:in `run': undefined local variable or method `droplet_ids' for #<Chef::Knife::DigitalOceanDropletDestroy:0x00000002a975c0> (NameError)
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.3.0/lib/chef/knife.rb:422:in `block in run_with_pretty_exceptions'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.3.0/lib/chef/local_mode.rb:39:in `with_server_connectivity'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.3.0/lib/chef/knife.rb:421:in `run_with_pretty_exceptions'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.3.0/lib/chef/knife.rb:214:in `run'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.3.0/lib/chef/application/knife.rb:142:in `run'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.3.0/bin/knife:25:in `<top (required)>'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/knife:23:in `load'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/knife:23:in `<main>'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by rudijs** [comment]
Also:

```
knife digital_ocean droplet destroy --all                                                                                          ⏎ ✱
ERROR: Warning all servers will be lost unless you exit with ctrl-c now!
14
ERROR: knife encountered an unexpected error
This may be a bug in the 'digital ocean droplet destroy' knife command or plugin
Please collect the output of this command with the `-VV` option before filing a bug report.
Exception: NameError: undefined local variable or method `droplet_ids' for #<Chef::Knife::DigitalOceanDropletDestroy:0x00000002022a68>
```

**Comment by rudijs** [comment]
@rmoriz I'd be happy to test any branches if you don't have an active DO account.

Cheers.

**Comment by rudijs** [comment]
Here's a little more output that may help:

```
knife digital_ocean droplet list                                                                                                     ✱
ID       Name     Size  Region       IPv4           Image      Status
5966659  redline  1gb   Amsterdam 3  188.166.85.18  14.04 x64  active

knife digital_ocean droplet destroy --server 5966659 -VV
INFO: Using configuration from /media/rudi/crypt2/projects/ride-share-market/devops/app/kitchen/.chef/knife.rb
/home/rudi/.rvm/gems/ruby-2.2.0/gems/knife-digital_ocean-2.4.1/lib/chef/knife/digital_ocean_droplet_destroy.rb:50:in `run': undefined local variable or method `droplet_ids' for #<Chef::Knife::DigitalOceanDropletDestroy:0x00000000a68d30> (NameError)
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.4.0/lib/chef/knife.rb:425:in `block in run_with_pretty_exceptions'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.4.0/lib/chef/local_mode.rb:39:in `with_server_connectivity'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.4.0/lib/chef/knife.rb:424:in `run_with_pretty_exceptions'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.4.0/lib/chef/knife.rb:215:in `run'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.4.0/lib/chef/application/knife.rb:142:in `run'
        from /home/rudi/.rvm/gems/ruby-2.2.0/gems/chef-12.4.0/bin/knife:25:in `<top (required)>'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/knife:23:in `load'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/knife:23:in `<main>'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
        from /home/rudi/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by rmoriz** [comment]
@rudijs can you please give the current master branch a try? PR #77 might have fixed the issue

**Comment by rudijs** [comment]
Hey Roland,

Yep that works.

I updated my Gemfile:

    gem "knife-digital_ocean", github: "rmoriz/knife-digital_ocean"
    bundle update knife-digital_ocean

Output snippet:

    Using knife-digital_ocean 2.4.1 from git://github.com/rmoriz/knife-digital_ocean.git (at master)

Test:

```
knife digital_ocean droplet list
ID       Name     Size   Region       IPv4            Image      Status
6002748  temp3    512mb  Amsterdam 3  188.166.93.190  14.04 x64  active

knife digital_ocean droplet destroy --server 6002748 -VV
INFO: Using configuration from /media/rudi/crypt2/projects/ride-share-market/devops/app/kitchen/.chef/knife.rb
Delete droplet with id: 6002748
OK
```

I was already on 2.4.1 so I guess you'll release 2.4.2 soon?

Thanks for your quick reply, this is a great Gem and very much appreicated!

**Comment by rmoriz** [comment]
Thanks for testing. I've just released v2.4.2.
